### ISSUE TITLE
Cannot get bucket configuration information from s3 data source

### ISSUE THREAD

**Issue by micah-spieler** [hashicorp/terraform-provider-aws#26054]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Since encryption at rest is a fundamental security requirement for companies like ours, it would be helpful to be able to retrieve the default encryption status of an s3 bucket. AWS s3 object encryption is available on the aws_s3_object resource and bucket policy configuration is available  on the aws_s3_bucket_policy , so I was surprised to see that default bucket encryption configuration is not available. Similarly, it might also be useful to be able to get public access settings for a bucket.

### New or Affected Resource(s)

* aws_s3_bucket
* aws_s3_bucket_policy

### References

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/s3_bucket

**Comment by jeremychauvet** [comment]
Working on it!

**Comment by nsergeev-nomihealth** [comment]
any update on this?

**Comment by kieran-lowe** [comment]
Yeah I'm really interested in this for validating configuration. Can we get this pushed out?

**Comment by daltschu22** [comment]
I have to shell out to the cli to get this data, having it part of the provider would be great!
